org/diceresearch/nebula/orchestrator.py

```diff
@@ -2,140 +2,141 @@
-                databasemanager.update_step(settings.results_table_name, settings.error_msg,
-                                            "There are no stances scores", identifier)
-            else:
-                tempjson = json.loads(STANCE_DETECTION_RESULT)
-                thread = threading.Thread(target=predictions.predict,
-                                              args=(tempjson, identifier))
-                thread.start()
-        except Exception as e:
-            logging.exception(e)
-            databasemanager.update_step(settings.results_table_name, settings.status, settings.error, identifier)
-            databasemanager.update_step(settings.results_table_name, settings.error_msg, str(e), identifier)
+        logging.debug("Stance detection")
+        thread = threading.Thread(target=cosine_similarity.calculate, args=(claims, identifier))
+        thread.start()
 
     elif next_stage == 6:
+        logging.debug('Query the trained model')
+        thread = threading.Thread(target=predictions.predict, args=(claims, identifier))
+        thread.start()
+
+    elif next_stage == 7:
+        logging.debug('Query the trained RNN model')
+        thread = threading.Thread(target=predictions.predict_mean, args=(claims, identifier))
+        thread.start()
+
+    elif next_stage == 8:
+        # run indicators if label is false
+        if veracity_label == settings.false_label:
+            indicators = run_indicator_check_text(input_text)
+            status = settings.completed
+            indicator_json = json.dumps(indicators, cls=SetEncoder)
+            databasemanager.update_json_step(settings.results_table_name, settings.results_indicator_check,
+                                        indicator_json, identifier)
+        else:
+            # otherwise show skipped status
+            status = settings.skipped
+
+        # update and increase the step
+        databasemanager.update_step(settings.results_table_name, settings.results_indicator_check_status,
+                                    status, identifier)
+        databasemanager.increase_the_stage(settings.results_table_name, identifier)
+        goNextLevel(identifier)
+
+    elif next_stage == 9:
+        # set status as done and send push notification to device token if existing
         databasemanager.update_step(settings.results_table_name, settings.status, settings.done, identifier)
         if REGISTRATION_TOKEN:
             notification.send_firebase_notification(REGISTRATION_TOKEN, NOTIFICATION_TITLE, NOTIFICATION_BODY)
             databasemanager.delete_from_column(settings.results_table_name,
                                                settings.results_notificationtoken_column_name)
-
-    elif next_stage == 7:
-        pass
-    elif next_stage == 8:
-        pass
-    elif next_stage == 9:
+    elif next_stage == 10:
         pass
     else:
         pass
```

Comment:
`elif next_stage == 10:` shouldn't exist since it is not used at all.

The `else:` part should throw an exception or at least log an error. If we end up in this block, something went wrong with our implementation. Typically, we want to know that :wink: